com.github.PopoutApps.popout3d.yml

```diff
@@ -0,0 +1,173 @@
+app-id: com.github.PopoutApps.popout3d
+
+runtime: org.freedesktop.Platform
+runtime-version: "20.08"
+sdk: org.freedesktop.Sdk
+
+command: popout3d
+rename-desktop-file: popout3d.desktop
+rename-icon: popout3d_logo
+copy-icon: true
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=host
+  
+modules:
+
+  - name: python3-pillow
+    buildsystem: simple
+    build-commands:
+      - pip3 install --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} Pillow Pillow-PIL
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/3c/7e/443be24431324bd34d22dd9d11cc845d995bcd3b500676bcf23142756975/Pillow-5.4.1.tar.gz
+        sha256: 5233664eadfa342c639b9b9977190d64ad7aca4edc51a966394d7e08e7f38a9f
+      - type: file
+        url: https://files.pythonhosted.org/packages/5d/e1/d298934cd78984a5e8d616945349ed66a74b9df0337fb1941ce76b8a2b7b/Pillow-PIL-0.1dev.tar.gz
+        sha256: d80ee30805f10bf898df9ce7495f46e88990e8f90b5c2b917722d87526252b2c
+
+  # Hugin deps
+  - shared-modules/glew/glew.json
+  - shared-modules/glu/glu-9.0.0.json
```

Comment:
Still the name doesn't match.... so it can't be found.

Reply:
Fixed. Have also changed Exiv2 module "sources" and am getting somewhere with tests of that on my PC.

Reply:
This has failed to build again. My bit for the Python stuff is okay, the problem lies with Hugin/align_image_stack. Any idea where I can get some help with this? I've posted a query on https://answers.launchpad.net/hugin/+question/695021 with no response so far.
I know this must be possible because there is a manifest here for Hugin
https://github.com/flathub/net.sourceforge.Hugin/blob/master/net.sourceforge.Hugin.json
and a Flatpak here
https://github.com/flathub/flathub/pull/1988

```
FAILED: src/hugin1/hugin/hugin 
..
..
view]+0x620): undefined reference to `wxGLCanvasX11::SwapBuffers()'
/usr/lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: src/hugin1/hugin/CMakeFiles/hugin.dir/GLViewer.cpp.o:(.data.rel.ro._ZTV10GLOverview[_ZTV10GLOverview]+0x638): undefined reference to `wxGLCanvas::GetXWindow() const'
/usr/lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: warning: creating DT_TEXTREL in a PIE
collect2: error: ld returned 1 exit status
[259/304] Building CXX object src/hugi...MakeFiles/PTBatcherGUI.dir/Batch.cpp.o
ninja: build stopped: subcommand failed.
Error: module align_image_stack: Child process exited with code 1

```

Reply:
I'm the maintainer of the Hugin flathub :-)

as I said: more recent version of Hugin works. What you can try is to build wxWidget the exact same way as in the Hugin manifest, and if that doesn't work, using an older version known to work.

Reply:
I have tried with all the URLs copied from the "sort of" manifest file on Launchpad. it doesn't work.

I appear to have come up against an unsolvable problem! When I build a Debian package with a dependency on Hugin, all I have to do is state that it "depends on" the Hugin Debian package. Flatpak does not allow me to state that it "depends” on the Hugin Flatpak. 

Flatpak requires me compile (I have written my app in an interpreted language and am not used to compilers) someone else's package (that I know nothing about) written in a language that I know nothing about (I have written my application in Python not C). I haven't got a snowball's chance in hell of achieving this.

I don't know which statements from the "sort of" manifest file on Launchpad would be valid and how to convert that format into a manifest.yml.

There might be a way - as you know Hubert, Hugin already has a Flatpak for which there must be a manifest.yml from which I might be able to copy the relevant bits. Could you give me a link to this please, otherwise this project is doomed...